workflows/custom-lint.R

```diff
@@ -30,88 +26,92 @@ doubleQuotesLinter = function(sourceFile) {
           line = line,
           ranges = list(c(col1, col2)))})})}
 
-newDefaults = with_defaults(
-  assignment_linter = NULL,
-  closed_curly_linter = NULL,
-  commented_code_linter = NULL,
-  cyclocomp_linter = NULL,
-  double_quotes_linter = doubleQuotesLinter,
-  line_length_linter(120),
-  object_name_linter = object_name_linter('camelCase'),
-  open_curly_linter = open_curly_linter(TRUE),
-  single_quotes_linter = NULL)
+
+getLintIgnore = function(path = 'lint_ignore.csv') {
+  lintIgnore = if (file.exists(path)) {
+    fread(path)
+  } else {
+    data.table(
+      filepath = as.character(), line_number = as.integer(),
+      message = as.character(), line = as.character())}
+  return(lintIgnore)}
+
 
 getLintDt = function(lintsFound, repository = NULL, branch = NULL) {
   if (is.null(repository)) {
-    dirIndexes = gregexpr('/', getwd())[[1]]
-    repository = substr(getwd(), dirIndexes[length(dirIndexes)] + 1, nchar(getwd()))}
+    dirIndexes = gregexpr('/', getwd())[[1L]]
+    repository = substr(getwd(), dirIndexes[length(dirIndexes)] + 1L, nchar(getwd()))}
 
   if (is.null(branch)) branch = 'main'
 
-  lfDt = unique(as.data.table(lintsFound), by = c('filename', 'line_number', 'message'))
+  lfDt = unique(as.data.table(lintsFound), by = c('filepath', 'line_number', 'message'))
```

Comment:
@jakejh I don't see anywhere you are changing this, but `filename` is the name that `lintr` generates. Do you want us to modify the name or just keep as is? I was keeping it this way just so I wasn't actually modifying anything `lintr` gave us back.